src/client/executor.rs

```diff
@@ -99,17 +138,21 @@ impl Client {
                     .topology
                     .handle_post_handshake_error(err.clone(), conn, server)
                     .await;
-                Err(err)
+                if !err.is_server_error() {
```

Comment:
From looking at [this section of the spec](https://github.com/mongodb/specifications/blob/master/source/retryable-reads/retryable-reads.rst#c-if-the-retry-attempt-fails), I think this should be checking for a retryable error as well, not just a server error:

>  If the retry failed due to another retryable error or some other error originating from the server, that error should be raised instead